profile.graphql

```diff
@@ -0,0 +1,48 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Thread monitor task create need data
+input ThreadMonitorTaskCreator {
+    # need to monitor service name
+    serviceId: ID!
+    # endpoint name to monitored under the special service.
+    endpointName: String!
+    # if null means task start now, or not means appoint task start time(timestamp)
+    startTime: Long
+    # thread monitor duration of this task(minute)
+    duration: Int!
+    # when the segment starts to execute, how long must it take before the monitor can be enbaled
+    minDurationThreshold: Int!
+    # when start monitor, time interval for each thread dumping the stack
+    dumpPeriod: Int!
+}
+
+# Thread monitor task create result
+input ThreadMonitorTaskCreateResult {
+    # task create is success
+    success: Boolean!
+
+    # get create error reason when not success
+    errorReason: String
+
+    # get data id when create success
+    id: String
+}
+
+extend type Query {
```

Comment:
Based on the GraphQL definition, https://graphql.org/learn/queries/#mutations, createTask should be in the Mutation.

Reply:
changed to extend type Mutation